homeassistant/components/home_connect/sensor.py

```diff
@@ -1,29 +1,95 @@
 """Provides a sensor for Home Connect."""
 
+from dataclasses import dataclass, field
 from datetime import datetime, timedelta
 import logging
 from typing import cast
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ENTITIES
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 import homeassistant.util.dt as dt_util
 
+from .api import ConfigEntryAuth, HomeConnectDevice
 from .const import (
+    ATTR_DEVICE,
     ATTR_VALUE,
+    BSH_EVENT_PRESENT_STATE_OFF,
     BSH_OPERATION_STATE,
     BSH_OPERATION_STATE_FINISHED,
     BSH_OPERATION_STATE_PAUSE,
     BSH_OPERATION_STATE_RUN,
+    COFFEE_EVENT_BEAN_CONTAINER_EMPTY,
+    COFFEE_EVENT_DRIP_TRAY_FULL,
+    COFFEE_EVENT_WATER_TANK_EMPTY,
     DOMAIN,
+    REFRIGERATION_EVENT_DOOR_ALARM_FREEZER,
+    REFRIGERATION_EVENT_DOOR_ALARM_REFRIGERATOR,
+    REFRIGERATION_EVENT_TEMP_ALARM_FREEZER,
 )
 from .entity import HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class HomeConnectSensorEntityDescription(SensorEntityDescription):
+    """Entity Description class for binary sensors."""
+
+    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
+    options: list[str] | None = field(
+        default_factory=lambda: ["confirmed", "off", "present"]
+    )
+    state_key: str
+    appliance_types: tuple[str, ...]
+
+
+SENSORS: tuple[HomeConnectSensorEntityDescription, ...] = (
+    HomeConnectSensorEntityDescription(
+        key="Door Alarm Freezer",
+        translation_key="alarm_sensor_freezer",
```

Comment:
It would be nice if we can refactor the integration to at some point use translations like normal so we can make the intergation more accessible for non native english people

Reply:
But in a followup

Reply:
I'll have to research this, would you know of a similar integration that does translations well?